<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowTitle">
   <string>Assistive Navigation â€” GUI (Designer)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="rootGrid">
    <property name="horizontalSpacing">
     <number>12</number>
    </property>
    <property name="verticalSpacing">
     <number>12</number>
    </property>
    <property name="margin">
     <number>12</number>
    </property>
    <!-- Left main area -->
    <item row="0" column="0">
     <widget class="QFrame" name="mainArea">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <layout class="QGridLayout" name="gridLeft">
       <property name="horizontalSpacing">
        <number>12</number>
       </property>
       <property name="verticalSpacing">
        <number>12</number>
       </property>
       <!-- Top videos -->
       <item row="0" column="0">
        <widget class="QLabel" name="videoMain">
         <property name="minimumSize">
          <size><width>480</width><height>300</height></size>
         </property>
         <property name="text">
          <string>videoMain (D435)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="videoRoad">
         <property name="minimumSize">
          <size><width>480</width><height>300</height></size>
         </property>
         <property name="text">
          <string>videoRoad (Webcam)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <!-- Bottom row (alerts + haptics + audio) -->
       <item row="1" column="0">
        <widget class="QGroupBox" name="alertsPanel">
         <property name="title">
          <string>alertsPanel</string>
         </property>
         <layout class="QVBoxLayout" name="alertsLayout">
          <item>
           <widget class="QListWidget" name="alertList"/>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QWidget" name="bottomRight">
         <layout class="QHBoxLayout" name="bottomRightLayout">
          <item>
           <widget class="QGroupBox" name="hapticsPanel">
            <property name="title">
             <string>hapticsPanel</string>
            </property>
            <layout class="QVBoxLayout" name="hapticsLayout">
             <item><widget class="QLabel" name="labelLeft"><property name="text"><string>Left motor</string></property></widget></item>
             <item><widget class="QProgressBar" name="pbLeft"><property name="value"><number>0</number></property></widget></item>
             <item><widget class="QLabel" name="labelRight"><property name="text"><string>Right motor</string></property></widget></item>
             <item><widget class="QProgressBar" name="pbRight"><property name="value"><number>0</number></property></widget></item>
             <item>
              <layout class="QHBoxLayout" name="hapticButtons">
               <item><widget class="QPushButton" name="btnTestL"><property name="text"><string>Test Left</string></property></widget></item>
               <item><widget class="QPushButton" name="btnTestR"><property name="text"><string>Test Right</string></property></widget></item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="audioPanel">
            <property name="title">
             <string>audioPanel</string>
            </property>
            <layout class="QVBoxLayout" name="audioLayout">
             <item><widget class="QListWidget" name="ttsQueue"/></item>
             <item>
              <layout class="QHBoxLayout" name="volumeRow">
               <item><widget class="QLabel" name="labelVolume"><property name="text"><string>Volume</string></property></widget></item>
               <item><widget class="QSlider" name="volSlider"><property name="orientation"><enum>Qt::Horizontal</enum></property><property name="minimum"><number>0</number></property><property name="maximum"><number>100</number></property><property name="value"><number>80</number></property></widget></item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <!-- Right sidebar -->
    <item row="0" column="1">
     <widget class="QFrame" name="rightArea">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <layout class="QVBoxLayout" name="rightLayout">
       <item>
        <widget class="QGroupBox" name="paramsPanel">
         <property name="title"><string>paramsPanel</string></property>
         <layout class="QVBoxLayout" name="paramsLayout">
          <item>
           <layout class="QHBoxLayout" name="safeRow">
            <item><widget class="QLabel" name="labelSafe"><property name="text"><string>Safe radius</string></property></widget></item>
            <item><widget class="QSlider" name="safeSlider"><property name="orientation"><enum>Qt::Horizontal</enum></property><property name="minimum"><number>50</number></property><property name="maximum"><number>400</number></property><property name="value"><number>150</number></property></widget></item>
           </layout>
          </item>
          <item><widget class="QLabel" name="labelClass"><property name="text"><string>Class filter</string></property></widget></item>
          <item>
           <layout class="QHBoxLayout" name="classRow">
            <item><widget class="QCheckBox" name="cbHuman"><property name="text"><string>Human</string></property><property name="checked"><bool>true</bool></property></widget></item>
            <item><widget class="QCheckBox" name="cbCar"><property name="text"><string>Car</string></property><property name="checked"><bool>true</bool></property></widget></item>
            <item><widget class="QCheckBox" name="cbBicycle"><property name="text"><string>Bicycle</string></property></widget></item>
            <item><spacer name="spClass"><property name="orientation"><enum>Qt::Horizontal</enum></property><property name="sizeHint"><size><width>40</width><height>20</height></size></property></spacer></item>
           </layout>
          </item>
          <item><widget class="QLabel" name="labelMode"><property name="text"><string>Mode</string></property></widget></item>
          <item>
           <layout class="QHBoxLayout" name="modeRow">
            <item><widget class="QRadioButton" name="rbOutdoor"><property name="text"><string>outdoor</string></property><property name="checked"><bool>true</bool></property></widget></item>
            <item><widget class="QRadioButton" name="rbIndoor"><property name="text"><string>indoor</string></property></widget></item>
            <item><spacer name="spMode"><property name="orientation"><enum>Qt::Horizontal</enum></property><property name="sizeHint"><size><width>40</width><height>20</height></size></property></spacer></item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="connectionPanel">
         <property name="title"><string>connectionPanel</string></property>
         <layout class="QVBoxLayout" name="connLayout">
          <item>
           <layout class="QHBoxLayout" name="connRow1">
            <item><widget class="QLabel" name="labelD435"><property name="text"><string>D435 depth</string></property></widget></item>
            <item><widget class="QLabel" name="lampD435"><property name="minimumSize"><size><width>14</width><height>14</height></size></property><property name="maximumSize"><size><width>14</width><height>14</height></size></property><property name="styleSheet"><string notr="true">background:#4caf50;border-radius:7px;</string></property></widget></item>
            <item><spacer name="sp1"><property name="orientation"><enum>Qt::Horizontal</enum></property><property name="sizeHint"><size><width>40</width><height>20</height></size></property></spacer></item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="connRow2">
            <item><widget class="QLabel" name="labelWebcam"><property name="text"><string>Webcam</string></property></widget></item>
            <item><widget class="QLabel" name="lampWebcam"><property name="minimumSize"><size><width>14</width><height>14</height></size></property><property name="maximumSize"><size><width>14</width><height>14</height></size></property><property name="styleSheet"><string notr="true">background:#ffc107;border-radius:7px;</string></property></widget></item>
            <item><spacer name="sp2"><property name="orientation"><enum>Qt::Horizontal</enum></property><property name="sizeHint"><size><width>40</width><height>20</height></size></property></spacer></item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="connRow3">
            <item><widget class="QLabel" name="labelServer"><property name="text"><string>Server</string></property></widget></item>
            <item><widget class="QLabel" name="lampServer"><property name="minimumSize"><size><width>14</width><height>14</height></size></property><property name="maximumSize"><size><width>14</width><height>14</height></size></property><property name="styleSheet"><string notr="true">background:#dc3545;border-radius:7px;</string></property></widget></item>
            <item><spacer name="sp3"><property name="orientation"><enum>Qt::Horizontal</enum></property><property name="sizeHint"><size><width>40</width><height>20</height></size></property></spacer></item>
           </layout>
          </item>
          <item><widget class="QPushButton" name="btnReconnect"><property name="text"><string>Reconnect</string></property></widget></item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="logTable">
         <property name="title"><string>logTable</string></property>
         <layout class="QVBoxLayout" name="logLayout">
          <item><widget class="QTableWidget" name="tbl"><column><number>3</number></column></widget></item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
